web-services/query/src/main/java/datawave/webservice/query/runner/RunningQuery.java

```diff
@@ -332,13 +331,16 @@ public ResultsPage next() throws Exception {
                 }
             }
         }
-        if (resultList.isEmpty()) {
-            return new ResultsPage();
-        } else {
-            return new ResultsPage(resultList, ((hitPageByteTrigger || hitPageTimeTrigger) ? ResultsPage.Status.PARTIAL : ResultsPage.Status.COMPLETE));
+        ResultsPage.Status status;
+        if (this.maxResults >= numResults) {
```

Comment:
My previous comment was "So if we have hit the max results, or it.hasNext() is false, then lets return COMPLETE in those cases.".  This if statement does not do that.  We "hit the max results" at lines 236 and 241.  In those cases we should return COMPLETE.  Also if !this.it.hasNext()  (i.e it.hasNext() is false) then we should also return COMPLETE.  Only after those checks should you return PARTIAL iff currentPageCount < maxPageSize.  So the code is something like this (provided hitMaxResults is set to true at lines 236 and 241):

if (hitMaxResults || !this.it.hasNext()) {
   status = ResultsPage.Status.COMPLETE;
} else {
   status = (currentPageCount < maxPageSize) ? ResultsPage.Status.PARTIAL : ResultsPage.Status.COMPLETE;
}
return new ResultsPage(resultsList, status);

Reply:
Oh, and BTW this code does not even compile.  Please ensure that all code compiles, passes the formatting, and all tests pass before pushing it up.  Thanks.

Reply:
I honstly don't know how I made the mistake of pushing this one. The correct one that i wrote was

  if (resultList.isEmpty()) {
            return new ResultsPage();
        } else if (iter.hasNext() && numResults < this.maxResults && currentPageCount < maxPageSize)
            return new ResultsPage(resultList, ResultsPage.Status.PARTIAL);
        else {
            return new ResultsPage(resultList, ResultsPage.Status.COMPLETE);
        }
    }
I am updating this PR now.